### PR TITLE
logging: Detect engine logging updates v1

### PR DESCRIPTION
**Pull request by jmtaylor90**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
- updated SCLog messages to be more consistent
- removed trailing space after log entry
-

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

### CODE REVIEW

File: src/detect-depth.c [hunk 1]
```diff
@@ -96,19 +96,19 @@ static int DetectDepthSetup (DetectEngineCtx *de_ctx, Signature *s, const char *
...
-                   "keyword set along with a fast_pattern:only;");
+                   "keyword set along with a fast_pattern:only;.");
         goto end;
     }
     if (str[0] != '-' && isalpha((unsigned char)str[0])) {
         SigMatch *bed_sm = DetectByteExtractRetrieveSMVar(str, s);
         if (bed_sm == NULL) {
             SCLogError(SC_ERR_INVALID_SIGNATURE, "unknown byte_extract var "
-                       "seen in depth - %s\n", str);
+                       "seen in depth - %s\n.", str);
```
Comment (victorjulien) [inline src/detect-depth.c]: the trailing \n should also be removed

File: src/detect-offset.c [hunk 2]
```diff
@@ -87,33 +87,33 @@ int DetectOffsetSetup (DetectEngineCtx *de_ctx, Signature *s, const char *offset
...
+                   "keyword set along with a fast_pattern:only;.");
         goto end;
     }
     if (str[0] != '-' && isalpha((unsigned char)str[0])) {
         SigMatch *bed_sm =
             DetectByteExtractRetrieveSMVar(str, s);
         if (bed_sm == NULL) {
             SCLogError(SC_ERR_INVALID_SIGNATURE, "unknown byte_extract var "
-                       "seen in offset - %s\n", str);
+                       "seen in offset - %s\n.", str);
```
Comment (victorjulien) [inline src/detect-offset.c]: trailing \n

**Comment by jmtaylor90** [comment]
> Could you rewrite the commit messages to not mention the filename? The filename is redundant as it's part of the diff itself. I prefer something 'detect/content: clean up logging output'

Will do.

File: src/detect-offset.c (hunk 2 above)
Comment (jmtaylor90) [reply src/detect-offset.c]: Will do.

File: src/detect-depth.c [hunk 3]
```diff
@@ -96,19 +96,19 @@ static int DetectDepthSetup (DetectEngineCtx *de_ctx, Signature *s, const char *
...
     if (cd->flags & DETECT_CONTENT_NEGATED && cd->flags & DETECT_CONTENT_FAST_PATTERN) {
         SCLogError(SC_ERR_INVALID_SIGNATURE, "can't have a relative "
-                   "negated keyword set along with a fast_pattern");
+                   "negated keyword set along with a fast_pattern.");
         goto end;
     }
     if (cd->flags & DETECT_CONTENT_FAST_PATTERN_ONLY) {
         SCLogError(SC_ERR_INVALID_SIGNATURE, "can't have a relative "
-                   "keyword set along with a fast_pattern:only;");
+                   "keyword set along with a fast_pattern:only;.");
```
[the remark this answers is not in the document]
Comment (jmtaylor90) [reply src/detect-depth.c]: to distinguish actual keywords, do we want to use single quotes around them? 
so in this case "keyword set along with a 'fast_pattern:only;'."?

[the remark this answers is not in the document]
Comment (jlucovsky) [reply src/detect-depth.c]: I think that improves readability.

**Comment by jmtaylor90** [comment]
continued in #3875

**Pull request closed without merging**